main.go

```diff
@@ -2,361 +2,34 @@ package main
-	config.RegisterDurationConfigVariable(720, &IdleTimeout, false, time.Second, []string{"IdleTimeout", "IdleTimeoutInSec"}...)
-	config.RegisterDurationConfigVariable(15, &gracefulShutdownTimeout, false, time.Second, "GracefulShutdownTimeout")
-	config.RegisterIntConfigVariable(524288, &MaxHeaderBytes, false, 1, "MaxHeaderBytes")
-
-	enterpriseToken = config.GetString("ENTERPRISE_TOKEN", enterpriseToken)
-}
-
-func Init() {
-	loadConfig()
-	pkgLogger = logger.NewLogger().Child("main")
-}
-
-func versionInfo() map[string]interface{} {
-	return map[string]interface{}{"Version": version, "Major": major, "Minor": minor, "Patch": patch, "Commit": commit, "BuildDate": buildDate, "BuiltBy": builtBy, "GitUrl": gitURL, "TransformerVersion": transformer.GetVersion(), "DatabricksVersion": misc.GetDatabricksVersion()}
-}
-
-func versionHandler(w http.ResponseWriter, _ *http.Request) {
-	version := versionInfo()
-	versionFormatted, _ := json.Marshal(&version)
-	_, _ = w.Write(versionFormatted)
-}
-
-func printVersion() {
-	version := versionInfo()
-	versionFormatted, _ := json.MarshalIndent(&version, "", " ")
-	fmt.Printf("Version Info %s\n", versionFormatted)
-}
-
-func startWarehouseService(ctx context.Context, application app.App) error {
-	return warehouse.Start(ctx, application)
-}
-
-func canStartServer() bool {
-	pkgLogger.Info("warehousemode ", warehouseMode)
-	return warehouseMode == config.EmbeddedMode || warehouseMode == config.OffMode || warehouseMode == config.PooledWHSlaveMode
-}
-
-func canStartWarehouse() bool {
-	return warehouseMode != config.OffMode
-}
-
-func runAllInit() {
-	admin.Init()
-	misc.Init()
-	db.Init()
-	diagnostics.Init()
-	backendconfig.Init()
-	warehouseutils.Init()
-	bigquery.Init()
-	clickhouse.Init()
-	archiver.Init()
-	destinationdebugger.Init()
-	pgnotifier.Init()
-	jobsdb.Init()
-	jobsdb.Init2()
-	jobsdb.Init3()
-	destinationconnectiontester.Init()
-	warehouse.Init()
-	warehouse.Init2()
-	warehouse.Init3()
-	warehouse.Init4()
-	warehouse.Init5()
-	warehouse.Init6()
-	validations.Init()
-	datalake.Init()
-	azuresynapse.Init()
-	mssql.Init()
-	postgres.Init()
-	redshift.Init()
-	snowflake.Init()
-	deltalake.Init()
-	transformer.Init()
-	webhook.Init()
-	batchrouter.Init()
-	batchrouter.Init2()
-	asyncdestinationmanager.Init()
-	batchrouterutils.Init()
-	dedup.Init()
-	eventschema.Init()
-	eventschema.Init2()
-	stash.Init()
-	transformationdebugger.Init()
-	processor.Init()
-	kafka.Init()
-	customdestinationmanager.Init()
-	routertransformer.Init()
-	router.Init()
-	router.InitRouterAdmin()
-	ratelimiter.Init()
-	sourcedebugger.Init()
-	gateway.Init()
-	apphandlers.Init()
-	apphandlers.Init2()
-	integrations.Init()
-	alert.Init()
-	multitenant.Init()
-	oauth.Init()
-	Init()
-}
-
 func main() {
 	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
-	exitCode := Run(ctx)
+	r := runner.New(runner.ReleaseInfo{
```

Comment:
Explain: moved running logic away from `main`, to the `runner` package, so that we can move integration test files in the packages that they belong.